tests/fault_injector/test_affected_element_does_not_exist.py

```diff
@@ -0,0 +1,182 @@
+                "end_tick": 100,
+                "description": "TrackSpeedLimitFault",
+                "affected_element_id": "12345678",
+                "new_speed_limit": 60,
+            },
+        ],
+    ],
+)
+class TestAffectedElementDoesNotExist:
+    """Test cases where the requested element for injecting a fault does not exist (in the simulation)"""
+
+    @recreate_db_setup
+    def setup_method(self):
+        pass
+
+    @pytest.fixture
+    def fault_configuration(self, table_class: BaseModel, object_as_dict: dict):
+        return table_class.create(**object_as_dict)
+
+    @pytest.fixture
+    def fault(
+        self,
+        fault_configuration: FaultConfiguration,
+        logger: Logger,
+        simulation_object_updater: SimulationObjectUpdatingComponent,
+        interlocking: IInterlockingDisruptor,
+        fault_type: Fault,
+    ):
+        return fault_type(
+            configuration=fault_configuration,
+            logger=logger,
+            simulation_object_updater=simulation_object_updater,
+            interlocking=interlocking,
+        )
+
+    def test_injection(self, tick, fault: Fault):
+        with pytest.raises(ValueError):
+            fault.inject_fault(tick)
+
+    def test_resolve_without_inject(self, tick, fault: Fault):
+        with pytest.raises(ValueError):
+            fault.resolve_fault(tick)
+
+
+class TestAffectedElementDoesNotExistScheduleBlockedFault:
+    """Tests cases where the requested element for injecting a schedule blocked fault does not exist (in the simulation)"""
+
+    @recreate_db_setup
+    def setup_method(self):
+        pass
+
+    @pytest.fixture
+    def schedule_blocked_fault_configuration(self):
+        return ScheduleBlockedFaultConfiguration(
+            **{
+                "start_tick": 1,
+                "end_tick": 100,
+                "description": "ScheduleBlockedFault",
+                "affected_element_id": "12345678",
+            }
+        )
+
+    @pytest.fixture
+    def schedule_blocked_fault(
+        self,
+        schedule_blocked_fault_configuration: ScheduleBlockedFaultConfiguration,
+        logger: Logger,
+        simulation_object_updater: SimulationObjectUpdatingComponent,
+        interlocking: IInterlockingDisruptor,
+        spawner: Spawner,
+    ):
+        return ScheduleBlockedFault(
+            configuration=schedule_blocked_fault_configuration,
+            logger=logger,
+            simulation_object_updater=simulation_object_updater,
+            interlocking=interlocking,
+            spawner=spawner,
+        )
+
+    def test_injection(self, tick, schedule_blocked_fault: ScheduleBlockedFault):
+        with pytest.raises(KeyError):
+            schedule_blocked_fault.inject_fault(tick)
+
+    def test_resolve_without_inject(
+        self, tick, schedule_blocked_fault: ScheduleBlockedFault
+    ):
+        with pytest.raises(KeyError):
+            schedule_blocked_fault.resolve_fault(tick)
```

Comment:
I think you are missing tests to resolve faults when the train does not exist anymore.

Reply:
What exactly do you mean? This tests schedules not trains